heltour/tournament/models.py

```diff
@@ -1583,7 +1584,19 @@ def save(self, *args, **kwargs):
                 old_black_setting = PlayerNotificationSetting.get_or_default(
                     player_id=self.initial_black_id, type='before_game_time', league=league)
                 old_black_setting.save()
-
+                
+            #set players available if game gets scheduled and either of them is set unavailable,
+            #do not set a player with a red card available, though.
+            if not SeasonPlayer.objects.filter(Q(player__id=self.white_id) & Q(season=self.get_round().season) & Q(games_missed__gte=2)).exists():
+                PlayerAvailability.objects.filter(
+                    (Q(player__id=self.white_id) & Q(round=self.get_round())
+                  )).update(is_available=True)
+            
+            if not SeasonPlayer.objects.filter(Q(player__id=self.black_id) & Q(season=self.get_round().season) & Q(games_missed__gte=2)).exists():
+                PlayerAvailability.objects.filter(
+                    (Q(player__id=self.black_id) & Q(round=self.get_round())
```

Comment:
I feel these two ifs can be quickly refactored to a single function and called twice, once for black and once for white. 

```python
def update_available_upon_scheduling(player_id):
    if not SeasonPlayer.objects.filter(player__id=player_id, season=self.get_round().season, games_missed__gte=2).exists():
        PlayerAvailability.objects.filter(player__id=player_id, round=self.get_round()).update(is_available=True)
```

Reply:
done